force-app/main/default/lwc/bulkServiceDeliveryUI/bulkServiceDeliveryUI.js

```diff
@@ -155,23 +157,39 @@ export default class BulkServiceDeliveryUI extends NavigationMixin(LightningElem
     handleRowError(event) {
         let errorIndex = event.detail.index;
         this.errors[errorIndex] = "error";
+        this.isAddEntryDisabled = false;
         this.renderErrors();
     }
 
-    handleClearError(event) {
-        let errorIndex = event.detail;
-        let result = this.serviceDeliveries.find(({ index }) => index === errorIndex);
-        result.hasSaved = true;
-        this.handleDeleteError(errorIndex);
-    }
-
     handleDeleteError(index) {
         if (this.errors[index]) {
             delete this.errors[index];
             this.renderErrors();
         }
     }
 
+    handleSubmit() {
+        this.isAddEntryDisabled = true;
+        this.isDoneDisabled = true;
+    }
+
+    handleSuccess(event) {
+        this.isAddEntryDisabled = false;
+
+        //If previously there was an error and then the record saved successfully the below piece of code removes that error
+        //by calling the handleDeleteError method
+        let errorIndex = event.target.index;
+        let result = this.serviceDeliveries.find(({ index }) => index === errorIndex);
+        result.hasSaved = true;
```

Comment:
Is result used anywhere?

Reply:
@jjbennett , result is not used anywhere but the `hasSaved` property is used while showing the toast. So we take a count of all the service deliveries where the `hasSaved` property is set to true. So this is needed here.

Reply:
Gotcha, thanks I missed that. What do you think about changing the variable name on line 181 to rowIndex and the variable on 182 to serviceDelivery? We have the comment on 179 which made me think this was specific to error handling but it looks like we are marking the row saved and removing errors if any where on that row?

Reply:
@jjbennett , Sorry for the confusing comment. I added it there . I can change the name of the variable.